### PR TITLE
Add TS type for `resolve` function

### PR DESCRIPTION
**Pull request by mattvague**
Noticed that the `resolve` function was missing a type definition. Amazing library @josdejong, nice work 😊

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -723,6 +723,13 @@ declare namespace math {
      */
     simplify: Simplify;
 
+     /**
+     *  Replaces variable nodes with their scoped values
+     * @param node Tree to replace variable nodes in
+     * @param scope Scope to read/write variables
+     */
+    resolve<TNode = MathNode>(node: TNode, scope: Record<string, any>): TNode;
+
```
Comment (gwhitney) [inline types/index.d.ts:732]: I am sure there is a reason, but why isn't this just
```
resolve(node: MathNode, scope: Record<string, any>): MathNode;
```
Thanks for explaining.

File: types/index.d.ts [hunk 2]
```diff
@@ -3773,6 +3781,12 @@ declare namespace math {
      */
     parser(): MathJsChain;
 
+    /**
+     *  Replaces variable nodes with their scoped values
+     * @param scope Scope to read/write variables
+     */
+    resolve(scope: Record<string, any>): MathJsChain;
+
```
Comment (gwhitney) [inline types/index.d.ts:3789]: Isn't the scope optional here, too?

File: src/function/algebra/resolve.js [hunk 3]
```diff
@@ -35,11 +35,12 @@ export const createResolve = /* #__PURE__ */ factory(name, dependencies, ({
...
-   * @param {Node} node
-   *     The expression tree to be simplified
+   * @template nodeOrNodes
+   * @param {nodeOrNodes} node
+   *     The expression tree (or trees) to be simplified
    * @param {Object} scope
    *     Scope specifying variables to be resolved
-   * @return {Node} Returns `node` with variables recursively substituted.
+   * @return {numOrStr extends Node[] ? Node[] : Node} Returns `nodeOrNodes` with variables recursively substituted.
    * @throws {ReferenceError}
```
Comment (gwhitney) [inline src/function/algebra/resolve.js:44]: I don't understand the significance of the identifier "numOrStr" here -- did you mean "nodeOrNodes"?

File: src/function/algebra/resolve.js [hunk 4]
```diff
@@ -35,11 +35,12 @@ export const createResolve = /* #__PURE__ */ factory(name, dependencies, ({
    *
    *     simplify, evaluate
    *
-   * @param {Node} node
-   *     The expression tree to be simplified
+   * @template nodeOrNodes
+   * @param {nodeOrNodes} node
+   *     The expression tree (or trees) to be simplified
```
Comment (gwhitney) [inline src/function/algebra/resolve.js:40]: I'd suggest keeping this less "fancy" and just writing @param { Node | Node[] } nodeOrNodes, but I am not picky either way as long as the web doc generation works -- please confirm that it does, thanks.

**Comment by gwhitney** [comment]
Sorry for all the separate comments, I always forget to start a review because I think "Oh, I just noticed one little thing, no need to start a review." Also, I think there should be a chain resolve test, for completeness' sake. And then I think that's everything I see, thanks for chasing all these tiny points down.

File: src/function/algebra/resolve.js (hunk 3 above)
Comment (mattvague) [reply src/function/algebra/resolve.js:44]: Shoot, yes I did. Bad copy n paste 🙄

File: types/index.ts [hunk 5]
```diff
@@ -1027,3 +1028,14 @@ Probability function examples
...
+/*
+Resolve examples
+*/
+{
+  const math = create(all, {});
+
+  expectTypeOf(math.resolve(math.parse('x + y'))).toMatchTypeOf<MathNode>()
+  expectTypeOf(math.resolve(math.parse('x + y'), { x: 0 })).toMatchTypeOf<MathNode>()
+  expectTypeOf(math.resolve([math.parse('x + y')], { x: 0 })).toMatchTypeOf<MathNode[]>()
+}
```
[the remark this answers is not in the document]
Comment (mattvague) [reply types/index.ts]: Added a VSCode setting for `maths` to always add newlines from now on 😉

File: types/index.d.ts (hunk 2 above)
Comment (mattvague) [reply types/index.d.ts:3789]: Yep, fixed and added a test

**Comment by mattvague** [comment]
>Sorry for all the separate comments, I always forget to start a review because I think "Oh, I just noticed one little thing, no need to start a review." Also, I think there should be a chain resolve test, for completeness' sake. And then I think that's everything I see, thanks for chasing all these tiny points down.

No problem at all, should be good now!

**Comment by josdejong** [comment]
Thanks Matt, looks good 👍

**Pull request merged**